server/shared/src/main/scala/org/http4s/server/middleware/HttpMethodOverrider.scala

```diff
@@ -147,15 +143,12 @@ object HttpMethodOverrider {
     def processRequest(req: Request[G]): F[Response[G]] =
       getUnsafeOverrideMethod(req).flatMap {
         case Some(m: String) => parseMethod.andThen(processRequestWithMethod(req, _)).apply(m)
-        case None => processRequestWithOriginalMethod(req)
+        case None => http(req)
       }
 
     Kleisli { (req: Request[G]) =>
-      config.overridableMethods
-        .contains(req.method)
-        .guard[Option]
-        .as(processRequest(req))
-        .getOrElse(processRequestWithOriginalMethod(req))
+      val isOverridden = config.overridableMethods.contains(req.method)
```

Comment:
As a fan of `fold` on `Boolean`, I'd rather do it like:
```
import mouse.boolean._

config.overridableMethods.contains(req.method).fold(processRequest(req), http(req))
```
Much more concise than using `Alternative` and `map` + `getOrElse` (and better by perf). But we haven't `Mouse` for now :( Though your variant probably is the best by perf.

Reply:
In my humble opinion, it is best to stick to the language Scala itself, and to the `if-then-else` operator that is taught on week 2 of any introductory programming course. Using a `fold` for `Boolean` seems to stretch that word a lot.

Reply:
Also not a fan of boolean fold.  Makes me pause and think whether true or false comes first, and my guess was wrong.